processor/awsapplicationsignalsprocessor/processor.go

```diff
@@ -0,0 +1,34 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package awsapplicationsignalsprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/awsapplicationsignalsprocessor"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+type awsapplicationsignalsprocessor struct{}
+
+func (ap *awsapplicationsignalsprocessor) StartMetrics(_ context.Context, _ component.Host) error {
```

Comment:
Is there any particular reason why `StartMetrics` and `StartTraces` are not combined as `Start` ?

Reply:
When a new processor is created via a factory (via `NewFactory()`), providing processor logic for traces and metrics is done separately (via [processor.WithTraces and processor.WithMetrics](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/870e99509a62f720601c93b6c4dcd5b4be29135b/processor/awsapplicationsignalsprocessor/factory.go#L31-L32))

So, `StartMetrics` and `StartTraces` are for the respective `processorhelper.NewMetricsProcessor` and `processorhelper.NewTracesProcessor`, for the respective metrics and traces processor.